### PR TITLE
Newslist page

### PR DESCRIPTION
**Pull request by RJCCrisler**
Made the news list page in Android.

### CODE REVIEW

File: src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/activities/WebViewActivity.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+        ViewUtil.setNavBarBackButton(this);
+        String url = getIntent().getExtras().getString("newslink");
+        startWebView(url);
+    }
+
+    // Add a progress dialog when loading
+    private void startWebView(String url) {
+
+//        webview.setWebViewClient(new WebViewClient() {
```
Comment (dvcrn) [inline src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/activities/WebViewActivity.java]: is all this commented out code needed?

File: src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/items/NewsItem.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    public View create(Context context) {
+        return LayoutInflater.from(context).inflate(R.layout.item_news, null);
+    }
+
+    @Override
+    public void display(View view) {
+        ButterKnife.bind(this, view);
+        Context context = view.getContext();
+
+        // Set the even on click
```
Comment (dvcrn) [inline src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/items/NewsItem.java]: `even` -> `event`

File: src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/activities/WebViewActivity.java (hunk 1 above)
Comment (RJCCrisler) [inline src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/activities/WebViewActivity.java]: well this was a code to add a progress dialog, but it's not used now so maybe I'll just delete it.

File: src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/screens/NewsListScreen.java [hunk 3]
```diff
@@ -15,14 +50,146 @@ public NewsListScreen(Context context, MblCarrier carrier, Map<String, Object> e
...
+                            currentCategory = categories[choice];
+                        }
+                        categoryButtonText.setText(currentCategory);
+                        dialog.dismiss();
+                        reload();
+                    }
+                });
+
 
+//        listDialog.setItems(
```
Comment (dvcrn) [inline src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/screens/NewsListScreen.java]: more unused code

File: src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/screens/NewsListScreen.java [hunk 4]
```diff
@@ -15,14 +50,146 @@ public NewsListScreen(Context context, MblCarrier carrier, Map<String, Object> e
...
+                        return compareResult;
+                    }
+                });
+
+
+                // Creating news items
+                List<MblUniversalItem> items = new ArrayList<>();
+                for (final News n : news) {
+                    items.add(new NewsItem(
+                            n,
```
Comment (dvcrn) [inline src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/screens/NewsListScreen.java]: maybe rename `n` to something else for readability? 😇 Though not a big deal

File: src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/activities/WebViewActivity.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        setContentView(R.layout.activity_webview);
+
+        ViewUtil.setNavBarBackButton(this);
+        String url = getIntent().getExtras().getString("newslink");
+        startWebView(url);
+    }
+
+    // Add a progress dialog when loading
+    private void startWebView(String url) {
+
```
Comment (datrjc) [inline src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/activities/WebViewActivity.java]: Why empty line here? :D

File: src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/db/models/News.java [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    private String category;
+
+    public News() {
+    }
+
+    public static List<News> all() {
+        return SQLite.select()
+                .from(News.class)
+                .where(News_Table.lang.eq(LangUtil.getLang()))
+                .orderBy(News_Table.date, false)
```
Comment (datrjc) [inline src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/db/models/News.java]: Should remove orderBy

File: src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/db/models/News.java [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    public static List<String> allCategories(){
+        List<News> newsList = SQLite.select(News_Table.category)
+                .distinct()
+                .from(News.class)
+                .where(News_Table.lang.eq(LangUtil.getLang()))
+                .queryList();
+        List<String> categories = new ArrayList<>();
+        //@TODO: need to localize ALL
```
Comment (datrjc) [inline src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/db/models/News.java]: Do this TODO

File: src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/screens/NewsListScreen.java [hunk 8]
```diff
@@ -15,14 +50,131 @@ public NewsListScreen(Context context, MblCarrier carrier, Map<String, Object> e
...
 
     @Override
     protected void reload() {
+        MblUtils.executeOnAsyncThread(new Runnable() {
+            @Override
+            public void run() {
+                // load from DB
+                List<News> news;
+
+                if (currentCategory.equals(getContext().getString(R.string.category_all ))) {
```
Comment (datrjc) [inline src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/screens/NewsListScreen.java]: You should define it somewhere
`getContext().getString(R.string.category_all )`

File: src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/screens/NewsListScreen.java [hunk 9]
```diff
@@ -15,14 +50,131 @@ public NewsListScreen(Context context, MblCarrier carrier, Map<String, Object> e
...
+                                        builder.append(BuildConfig.API_SERVER)
+                                                .append("/api/v2/owner/news/bridge?link=")
+                                                .append(URLEncoder.encode(n.getLink(), "UTF-8"));
+                                    } catch (UnsupportedEncodingException e) {
+                                        e.printStackTrace();
+                                    }
+
+                                    webIntent.putExtra("newslink", builder.toString());
+                                    getContext().startActivity(webIntent);
+                                    ((BaseActivity) getContext()).overridePendingTransition(R.anim.pull_in_right, R.anim.push_out_left);
```
Comment (datrjc) [inline src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/screens/NewsListScreen.java]: we should do this animation for all activities

File: src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/screens/NewsListScreen.java [hunk 10]
```diff
@@ -15,14 +50,131 @@ public NewsListScreen(Context context, MblCarrier carrier, Map<String, Object> e
...
+        super.onEvent(sender, name, args);
+        if (name == Event.SYNCHRONIZATION_FINISHED){
+            reload();
+        }
+    }
+
+    @OnClick(R.id.category_button)
+    void onCategoryButtonClick() {
+
+        List<String> categoryList = News.allCategories();
```
Comment (datrjc) [inline src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/screens/NewsListScreen.java]: We should put this on async thread

File: src/WealthParkOwner/app/src/main/res/layout/item_news.xml [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+            android:orientation="vertical"
+            android:gravity="bottom"
+            android:background="@drawable/gradation"
+            android:padding="15dp">
+
+            <TextView
+                android:id="@+id/title_text"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:textColor="#fff"
```
Comment (datrjc) [inline src/WealthParkOwner/app/src/main/res/layout/item_news.xml]: font size is not correct

**Comment by RJCCrisler** [comment]
Updated code

File: src/WealthParkOwner/app/src/main/res/layout/include_nav_bar.xml [hunk 12]
```diff
@@ -43,4 +42,29 @@
...
+        android:layout_gravity="center"
+        android:gravity="center"
+        android:paddingLeft="10dp"
+        android:paddingRight="10dp"
+        android:orientation="horizontal">
+
+        <TextView
+            android:id="@+id/category_button_text"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
```
Comment (datrjc) [inline src/WealthParkOwner/app/src/main/res/layout/include_nav_bar.xml]: This TextView needs textColor and textSize

File: src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/items/NewsItem.java [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+        Context context = view.getContext();
+
+        // Set the event on click
+        view.setOnClickListener(onClickListener);
+
+        try {
+            int lastIndex = news.getPicture().lastIndexOf('/') + 1;
+            final String imgUrl = (news.getPicture().substring(0,lastIndex)+ URLEncoder.encode(news.getPicture().substring(lastIndex), "UTF-8"));
+
+            Picasso.with(context)
```
Comment (datrjc) [inline src/WealthParkOwner/app/src/main/java/com/wealthpark/owner/items/NewsItem.java]: App will crash if `news.getPicture()` is empty. Need to catch all exceptions and show placeholder image when exception happends

**Comment by datrjc** [comment]
I think it is ok. thanks

**Comment by gabuchan** [comment]
getting late to the party but: :+1:

**Pull request merged**